### ISSUE TITLE
Remote validation incorrectly relies on previous value even after resetForm()

### ISSUE THREAD

**Issue by dashfwd** [jquery-validation/jquery-validation#312]
I have a form that does server-side (remote) validation on a "name" field, to make sure it isn't a duplicate.

I fill out the form on the page with the name of "Fred", the server-side validation runs, and no errors occur, so I insert a row into the database for Fred via an AJAX call.  I never leave the page.

Now the user clicks the "Add" button again (on the same page), and the user types in the exact same name ("Fred").  Here, server-side validation will never run because of this line in the "remote" method of the plugin:

```
        if ( previous.old === value ) {
            return previous.valid;
        }
```

In other words, it will completely skip the validation check if that value was previously valid.  Note that calling resetForm() has no impact, it will continue to think that it does not need to be validated.

My temporary work around is to refresh the entire page, but it would be nice if the validation plugin could handle this case, preferably if I reset the form.

Note - this does NOT seem to be the same issue as #201, "A bug when field value change using validate remote".
